app/controllers/idv/hybrid_mobile/socure/document_capture_controller.rb

```diff
@@ -50,17 +49,66 @@ def show
         end
 
         def update
+          return if wait_for_result?
+
           result = handle_stored_result(
             user: document_capture_session.user,
             store_in_session: false,
           )
+          # TODO: new analytics event?
+          analytics.idv_doc_auth_document_capture_submitted(
+            **result.to_h.merge(analytics_arguments),
+          )
 
           if result.success?
             redirect_to idv_hybrid_mobile_capture_complete_url
           else
             redirect_to idv_hybrid_mobile_socure_document_capture_url
           end
         end
+
+        private
+
+        def wait_for_result?
+          return false if stored_result.present?
+
+          # If the stored_result is nil, the job fetching the results has not completed.
+          analytics.idv_doc_auth_document_capture_polling_wait_visited(**analytics_arguments)
+          if wait_timed_out?
+            # flash[:error] = I18n.t('errors.doc_auth.polling_timeout')
+            # TODO: redirect to try again page LG-14873/14952/15059
+            render plain: 'Technical difficulties!!!', status: :ok
+          else
+            @refresh_interval =
+              IdentityConfig.store.doc_auth_socure_wait_polling_refresh_max_seconds
+            render 'idv/socure/document_capture/wait'
+          end
+
+          true
+        end
+
+        def wait_timed_out?
+          if session[:socure_docv_wait_polling_started_at].nil?
+            session[:socure_docv_wait_polling_started_at] = Time.zone.now.to_s
+            return false
+          end
+          start = DateTime.parse(session[:socure_docv_wait_polling_started_at])
+          timeout_period =
+            IdentityConfig.store.doc_auth_socure_wait_polling_timeout_minutes.minutes || 2.minutes
+          start + timeout_period < Time.zone.now
+        end
+
+        def analytics_arguments
+          {
+            flow_path: 'hybrid',
+            step: 'socure_hybrid_document_capture',
+            analytics_id: 'Doc Auth',
+            redo_document_capture: nil,
```

Comment:
i don't think we have a clearway to determine this value for the update but could be nice for the show action, i think that's why we don't currently log it in LN hybrid document capture ... perhaps we could add [this method](https://github.com/18F/identity-idp/blob/fa0524f750ddcc704d84f714fcca70f9c8b959cb/app/controllers/idv/hybrid_mobile/document_capture_controller.rb#L78) and then either return `true` or `nil` ... or just remove it? 😅